app/utils/cookies.js

```diff
@@ -133,6 +138,43 @@ function deleteSessionVariable (req, key) {
   }
 }
 
+/**
+ * @param {Request} req
+ * @returns {string}
+ */
+function findSessionChargeToDelete (req) {
+  const now = epochSecondsNow()
+  const yesterday = now - 86400
+  const ninetyMinutesAgo = now - 5400
+
+  const chargesSortedByAccessedDate = getChargesOnSession(req)
+    .map(chargeId => {
+      return {
+        chargeId,
+        state: getSessionChargeState(req, chargeId)
+      }
+    })
+    .sort(({ state: a }, { state: b }) => a.accessedAt - b.accessedAt)
+
+  logger.info(`User session contains ${chargesSortedByAccessedDate.length} charges`)
+
+  const leastRecentlyAccessedChargeCreatedMoreThanTwentyFourHoursAgo = chargesSortedByAccessedDate.find(charge => charge.state.createdAt < yesterday)
+
+  if (leastRecentlyAccessedChargeCreatedMoreThanTwentyFourHoursAgo) { return leastRecentlyAccessedChargeCreatedMoreThanTwentyFourHoursAgo.chargeId }
+
+  const leastRecentlyAccessedTerminalCharge = chargesSortedByAccessedDate.find(charge => charge.state.isTerminal)
+
+  if (leastRecentlyAccessedTerminalCharge) { return leastRecentlyAccessedTerminalCharge.chargeId }
+
+  const leastRecentlyAccessedChargeCreatedMoreThanNinetyMinutesAgo = chargesSortedByAccessedDate.find(charge => charge.state.createdAt < ninetyMinutesAgo)
+
+  if (leastRecentlyAccessedChargeCreatedMoreThanNinetyMinutesAgo) { return leastRecentlyAccessedChargeCreatedMoreThanNinetyMinutesAgo.chargeId }
+
+  logger.info('No preferred charges found for deletion on session, defaulting to least recently accessed charge')
```

Comment:
Would logging the accessed timestamp be useful here?

Reply:
I don't think it's necessary, we can add more detail to the logs if we see this cropping up a lot. As it stands this should be quite rare